bokehjs/test/unit/models/tools/inspectors/hover_tool.ts

```diff
@@ -56,4 +58,170 @@ describe("HoverTool", () => {
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">swatch &amp; hex (unknown): </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">__colors unknown</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">hex, swatch (unknown): </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">__colors unknown</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">swatch, hex (unknown): </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">__colors unknown</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">hex (unknown): </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">__colors unknown</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">swatch (unknown): </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">__colors unknown</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">foo: </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">abcd</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+  <div class="bk" style="display: table-row;">
+    <div class="bk bk-tooltip-row-label" style="display: table-cell;">bar: </div>
+    <div class="bk bk-tooltip-row-value" style="display: table-cell;">
+      <span class="bk" data-value="">-1</span>
+      <span class="bk bk-tooltip-color-block" data-swatch="" style="display: none;"> </span>
+    </div>
+  </div>
+</div>
+`
+    expect(el!.outerHTML).to.be.equal(html.trim().split("\n").map((s) => s.trim()).join(""))
```

Comment:
I didn't realize DOM inspection like this was possible now outside the selenium tests. This is great (and a number of those tests could be simplified to not rely on on selenium now)

Reply:
This is possible since we started using a real browser for unit testing. However, this is still not ideal. I will add a proper HTML node comparator, so that all this trimming, at al. would not be needed.